draft-zsf.md

```diff
@@ -0,0 +1,154 @@
+6. **Consensus Continuity During Lulls:** Rate-limited payouts from the Sustainability Fund help maintain blockchain security during periods of low activity by utilizing savings from high activity periods, whereas direct miner rewards tied to short-term activity may favor miners with significant savings reserves or access to credit, reducing mining competitiveness and overall ecosystem certainty about mining capacity.
+7. **Mitigate Payment-to-Self Vulnerabilities:** The risk of miners benefiting from "spoofing" transactions and the desire for a fixed eventual supply goal lead to considering alternatives like the Sustainability Fund, which pays out fees over a long enough time horizon to prevent recouping costs, allowing Zcash to remain closer to its supply goal and offering an alternative to other protocol designs that destroy or burn funds.
+8. **Recovery of "Soft-Burned" Funds:** In some instances, such as miners not claim all available rewards, some ZEC goes unaccounted for though not formally burned. This proposal would recover it through the `ZSF_BALANCE` mechanism described below.
+
+_Disclaimer 1: Long term success depends on the specific mechanisms of deposits, the quantities involved, and broader economic considerations. For such as if the payout rate is both sufficient and not excessive enough to threaten sustainability._
+
+# Specification
+
+In practice, The Zcash Sustainability Fund is a single global balance maintained by the node state and contributed to via a single transaction field. This provides the economic and security support described in the motivation section, while also importantly keeping the fund payouts extremely simple to describe and implement.
+
+The two modifications are:
+1. The re-accounting of unmined ZEC as a node state field called `ZSF_BALANCE`
+2. The addition of a transaction field called `ZSF_DEPOSIT`
+
+Please note that a **network upgrade is required** for this work to be fully implemented.
+
+## `ZSF_BALANCE`
+
+Upon activation at height `H`, the ZEC issuance mechanism is permanently suspended, the value of `ZSF_BALANCE[H]` is set to the remainder of unissued ZEC. This must be done before the `Block Rewards Payout Rule` is enforced for the block at height `H`. **This is not a pre-mine.** This is simply a re-accounting of unissued ZEC that comes with the benefit of other possible sources of funding.
+
+Consensus nodes are then required to track new per-block state:
+
+- `ZSF_BALANCE[H] : u64 [zatoshi]`
+
+The state is a single 64 bit integer (representing units of `zatoshi`) at any given block height, ``H``, representing the Sustainability Fund balance at that height, ``H``. The `ZSF_BALANCE` can be calculated using the following formula:
+
+`TOTAL ZEC TO EXIST (MAX_MONEY) - CLAIMED BLOCK SUBSIDIES OF PAST BLOCKS + SUM OF ALL ZSF DEPOSITS FROM PAST TRANSACTIONS`
```

Comment:
This should be made absolutely precise:

`ZsfBalanceAfter(height) = MAX_MONEY + sum_{h = 0}^{height} (ZsfDeposit(height) + Unclaimed(height) - BlockSubsidy(height))`

where `Unclaimed(height)` is the portion of the block subsidy that is unclaimed for the block at the given height. (This can also be written more precisely but don't worry about that for now.)